lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/TaskStopExceptionTest.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                // submit the first Task.
+                value.SubmitTask(TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, "TaskID")
+                        .Set(TaskConfiguration.Task, GenericType<TaskStopExceptionTask>.Class)
+                        .Set(TaskConfiguration.OnTaskStop, GenericType<TaskStopExceptionHandler>.Class)
+                        .Build());
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                // Should only receive one CompletedTask, as validated.
+                Logger.Log(Level.Info, CompletedTaskReceived);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                // Check that Exceptions are deserialized correctly.
+                var ex = value.AsError();
+                if (ex == null)
+                {
+                    throw new Exception("Exception was not expected to be null.");
+                }
+
+                var taskStopEx = ex as TaskStopExceptionTestException;
+
+                if (taskStopEx == null)
+                {
+                    throw new Exception("Expected Exception to be of type TaskStopExceptionTestException, but instead got type " + ex.GetType().Name);
+                }
+
+                if (taskStopEx.Message != TaskStopExceptionMessage)
+                {
+                    throw new Exception(
+                        "Expected message to be " + TaskStopExceptionMessage + " but instead got " + taskStopEx.Message + ".");
+                }
+
+                Logger.Log(Level.Info, FailedTaskReceived);
+
+                // Submit the new Task to verify that the original Context accepts new Tasks.
+                value.GetActiveContext().Value.SubmitTask(
+                    TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, "TaskID")
+                        .Set(TaskConfiguration.Task, GenericType<TaskStopExceptionResubmitTask>.Class)
+                        .Build());
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class TaskStopExceptionTask : LoggingTask
+        {
+            [Inject]
+            private TaskStopExceptionTask() 
+                : base(InitialTaskMessage)
+            {
+            }
+        }
+
+        private sealed class TaskStopExceptionResubmitTask : LoggingTask
+        {
+            [Inject]
+            private TaskStopExceptionResubmitTask() 
+                : base(ResubmitTaskMessage)
+            {
+            }
+        }
+
+        private sealed class TaskStopExceptionHandler : ExceptionThrowingHandler<ITaskStop>
```

Comment:
It is a stop handler which throw exception. May be rename it as TaskStopHandlerWithException